readers/api/endpoint_test.go

```diff
@@ -125,722 +127,860 @@ func TestReadAll(t *testing.T) {
-			status: http.StatusOK,
+			desc:         "read page with from/to as thing",
+			url:          fmt.Sprintf("%s/channels/%s/messages?from=%f&to=%f", ts.URL, chanID, messages[19].Time, messages[4].Time),
+			key:          thingToken,
+			authResponse: &magistrala.AuthorizeRes{Authorized: true},
+			status:       http.StatusOK,
 			res: pageRes{
-				Total:    uint64(len(messages[5:20])),
-				Messages: messages[5:15],
+				PageMetadata: readers.PageMetadata{Limit: 10, Format: "messages", From: messages[19].Time, To: messages[4].Time},
+				Total:        uint64(len(messages[5:20])),
+				Messages:     messages[5:15],
 			},
 		},
 		{
-			desc:   "read page with aggregation as thing",
-			url:    fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX", ts.URL, chanID),
-			key:    thingToken,
-			status: http.StatusBadRequest,
+			desc:         "read page with aggregation as thing",
+			url:          fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX", ts.URL, chanID),
+			key:          thingToken,
+			authResponse: &magistrala.AuthorizeRes{Authorized: true},
+			status:       http.StatusBadRequest,
 		},
 		{
-			desc:   "read page with interval as thing",
-			url:    fmt.Sprintf("%s/channels/%s/messages?interval=10h", ts.URL, chanID),
-			key:    thingToken,
-			status: http.StatusOK,
+			desc:         "read page with interval as thing",
+			url:          fmt.Sprintf("%s/channels/%s/messages?interval=10h", ts.URL, chanID),
+			key:          thingToken,
+			authResponse: &magistrala.AuthorizeRes{Authorized: true},
+			status:       http.StatusOK,
 			res: pageRes{
-				Total:    uint64(len(messages)),
-				Messages: messages[0:10],
+				PageMetadata: readers.PageMetadata{Limit: 10, Format: "messages", Interval: "10h"},
+				Total:        uint64(len(messages)),
+				Messages:     messages[0:10],
 			},
 		},
 		{
-			desc:   "read page with aggregation and interval as thing",
-			url:    fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX&interval=10h", ts.URL, chanID),
-			key:    thingToken,
-			status: http.StatusBadRequest,
+			desc:         "read page with aggregation and interval as thing",
+			url:          fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX&interval=10h", ts.URL, chanID),
+			key:          thingToken,
+			authResponse: &magistrala.AuthorizeRes{Authorized: true},
+			status:       http.StatusBadRequest,
 		},
 		{
-			desc:   "read page with aggregation, interval, to and from as thing",
-			url:    fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX&interval=10h&from=%f&to=%f", ts.URL, chanID, messages[19].Time, messages[4].Time),
-			key:    thingToken,
-			status: http.StatusOK,
+			desc:         "read page with aggregation, interval, to and from as thing",
+			url:          fmt.Sprintf("%s/channels/%s/messages?aggregation=MAX&interval=10h&from=%f&to=%f", ts.URL, chanID, messages[19].Time, messages[4].Time),
+			key:          thingToken,
+			authResponse: &magistrala.AuthorizeRes{Authorized: true},
```

Comment:
Use boolean instead of full `AuthroizeRes`. Do the same for all the other cases.